ports/qemu/Makefile

```diff
@@ -19,16 +19,20 @@ QSTR_DEFS = qstrdefsport.h
 MICROPY_ROM_TEXT_COMPRESSION ?= 1
 
 ifeq ($(QEMU_ARCH),arm)
-FROZEN_MANIFEST ?= "require('unittest'); freeze('test-frzmpy')"
+QEMU_GC_HEAP_SIZE ?= 100
+FROZEN_MANIFEST ?= "require('unittest'); freeze('test-frzmpy', ('frozen_asm_thumb.py', 'frozen_const.py', 'frozen_viper.py', 'native_frozen_align.py'))"
 endif
 ifeq ($(QEMU_ARCH),riscv32)
-FROZEN_MANIFEST ?= "require('unittest'); freeze('test-frzmpy', ('frozen_const.py', 'frozen_viper.py', 'native_frozen_align.py'))"
+QEMU_GC_HEAP_SIZE ?= 120
```

Comment:
Can we just increase the heap size to 120k for all boards/archs?  That would keep it simple (if it's possible to do, if the boards all have enough RAM for that, even though it's all virtual).

I would prefer to keep it simple and use the one size.  But if you think it's needed (or a much better idea) to have this configurable, then I suggest a different name for the variable.  The prefix `QEMU_` indicates that it's a qemu-related config option, when it's actually a MicroPython option.  So maybe just `GC_HEAP_SIZE`, or `MICROPY_GC_HEAP_SIZE`.

Reply:
Sharing the same default for both Arm and RV32 could work but then that value would need to be changed once more for #15931, to a value that may not reflect the reality of the physical boards the code under test will run on.  Unless RV64 ends up with a different default, and at that point it's pretty much the same as having a per-arch default.

Besides fixing the RV32 tests (for which I still have no idea why that happens), I've extensively used this feature to see how my code behaves in low-memory conditions, due to a simulated testing scenario, or due to a fragmented heap after long periods of (simulated) execution time.  I found it much faster to script a command line argument to `make` than messing with `CFLAGS`.

W.r.t. the prefix, you've got valid points, but on the other hand this flag is specific to the QEMU test runner.  I'd say that keeping the prefix sort of makes it more explicit that it is strictly a testing-only configuration flag, but I'm not 100% against any changes on that front.

Reply:
No worries, I missed the bit about those other two ports - I rarely interact with minimal or with zephyr, if at all.

I've updated the user-facing name to `MICROPY_HEAP_SIZE` in both the code and the port's README file.